### PR TITLE
PARQUET-1601: Add zstd support to parquet-cli to-avro

### PR DESCRIPTION
**Pull request by sekikn**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-1601
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

  - testToAvroCommandWithGzipCompression
  - testToAvroCommandWithSnappyCompression
  - testToAvroCommandWithZstdCompression
  - testToAvroCommandWithInvalidCompression

In addition, I confirmed an output file generated by to-avro was properly compressed by zstd as follows:

```
$ java -cp 'target/dependency/*:target/*' org.apache.parquet.cli.Main to-avro sample.parquet -o sample.avro --compression-codec zstd
$ java -jar ~/avro-tools-1.9.0.jar getmeta sample.avro 

(snip)

avro.codec	zstandard
```

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -106,6 +106,7 @@
     <opencsv.version>2.3</opencsv.version>
     <jcommander.version>1.35</jcommander.version>
     <commons-codec.version>1.10</commons-codec.version>
+    <zstd-jni.version>1.3.8-6</zstd-jni.version>
```
Comment (Fokko) [inline pom.xml]: Maybe set this to `1.4.0-1` right away.

Comment (sekikn) [reply pom.xml]: Thanks for the comment @Fokko! [Avro 1.9.0 seems to depend on zstd-jni 1.3.8-6](https://github.com/apache/avro/blob/release-1.9.0/lang/java/pom.xml#L65), so I reconciled them. Should I update it?

Comment (Fokko) [reply pom.xml]: Good point, maybe leave it as is for now to avoid conflicts

Comment (Fokko) [reply pom.xml]: https://github.com/apache/avro/pull/549

**Comment by sekikn** [comment]
Rebased on master and resolved conflicts.

**Pull request merged**